launcher/minecraft/mod/tasks/BasicFolderLoadTask.h

```diff
@@ -49,10 +49,16 @@ class BasicFolderLoadTask : public Task {
             connect(this, &Task::finished, this->thread(), &QThread::quit);
 
         m_dir.refresh();
+        QStringList names;
         for (auto entry : m_dir.entryInfoList()) {
             auto resource = m_create_func(entry);
-            resource->moveToThread(m_thread_to_spawn_into);
-            m_result->resources.insert(resource->internal_id(), resource);
+            if (names.contains(resource->name())) {
+                resource->destroy();
+            } else {
+                names << resource->name();
+                resource->moveToThread(m_thread_to_spawn_into);
+                m_result->resources.insert(resource->internal_id(), resource);
+            }
```

Comment:
as simple and foolproof as this seems I'm not sure this is either the correct place for this logic or even the correct logic.

as is, two resources with the same name (even if they are different versions or even different resources) cause conflicts, but silently deleting the second resource of the same name found seems wrong. renaming it to something like `<name>.duplicateN` and highlight it in the list seems better.

in the case of PolyMC/#550 and the downloading a duplicate mod. is seems like better logic to explicitly look for the expected `<name>.disabled` file during the download and handling that case in the download logic.

Reply:
Just to not forget: moving the logic to download mod will not suffice.
You can still add the same mod from external sources(and also the Add Mod button).
And this also affects all the other resources so I would love to implement it as low as possible to not have to implement it for each view.
Other than that your suggestion with duplicate suffix I will make it.

Reply:
Now it renames the files instead of removing them